lib/internal/Magento/Framework/DB/Ddl/Table.php

```diff
@@ -427,13 +428,23 @@ public function addForeignKey($fkName, $column, $refTable, $refColumn, $onDelete
             default:
                 $onDelete = self::ACTION_NO_ACTION;
         }
+        switch ($onUpdate) {
+            case self::ACTION_CASCADE:
+            case self::ACTION_RESTRICT:
+            case self::ACTION_SET_DEFAULT:
+            case self::ACTION_SET_NULL:
+                break;
+            default:
+                $onUpdate = self::ACTION_NO_ACTION;
+        }
 
         $this->_foreignKeys[$upperName] = [
             'FK_NAME' => $fkName,
             'COLUMN_NAME' => $column,
             'REF_TABLE_NAME' => $refTable,
             'REF_COLUMN_NAME' => $refColumn,
-            'ON_DELETE' => $onDelete
+            'ON_DELETE' => $onDelete,
+            'ON_UPDATE' => $onUpdate
```

Comment:
You're right. This needs a lot more work, but unfortunately I don't have time. Since this is a bug I hope to see it fixed soon on the main branch though.